framework/src/systems/AuxiliarySystem.C

```diff
@@ -736,12 +736,20 @@ AuxiliarySystem::computeMortarNodalVars(const ExecFlagType type)
       {
         PARALLEL_TRY
         {
-          ComputeMortarNodalAuxBndThread<AuxKernel> mnabt(_fe_problem, mortar_nodal, bnd_id, index);
-          Threads::parallel_reduce(bnd_nodes, mnabt);
-          solution().close();
-          _sys.update();
+          try
+          {
+            ComputeMortarNodalAuxBndThread<AuxKernel> mnabt(
+                _fe_problem, mortar_nodal, bnd_id, index);
+            Threads::parallel_reduce(bnd_nodes, mnabt);
+          }
+          catch (MooseException & e)
+          {
+            _fe_problem.setException(e.what());
+          }
         }
         PARALLEL_CATCH;
+        solution().close();
+        _sys.update();
```

Comment:
`solution().close();` will perform MPI communication, however it will never be reached if a libMesh exception is thrown from within `ComputeMortarNodalAuxBndThread`